### PR TITLE
Insert bug data form into database pt2

### LINKED ISSUE #4: Create submit page for bug report
Issue 2B

- Estimated Effort: 3 days
- Dependencies: Bug report form must be created first
- Initial Assignments: [Ellen Fang](mailto:ellenf@andrew.cmu.edu)
- Milestones: Completed in Sprint 2
- Acceptance Criteria:
  -  Form has a submit button and prints out the collected data (for testing purposes)

### PR DESCRIPTION
**Pull request by Sebastian0011**
Resolves https://github.com/CMU-313/fall23-nodebb-formula1/issues/4 https://github.com/CMU-313/fall23-nodebb-formula1/issues/6

(Branches off of #28 )
- Creates jQuery file that creates object with form data and posts it to api
- Write controller contains new function submitData
- Uses the bug controller's create function to insert into the database
- Translates src/bugs/create.js to ts
- Creates bid (bug id) as unique identifier for bug reports
- Removed the timestamp from the bug report (causing issues with the post / fetch)

Test:

- pull code
- run ./nodebb build && ./nodebb start
- go to bug report form page by clicking on the bug icon in nav
- Fill out form (name, email, date, and description are required)
- check that the values of the form are printed in ./nodebb log

### CODE REVIEW

File: public/src/client/bugs.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+define('forum/bug-report', [
+    'api', 'alerts',
+], function (api, alerts) {
+    const Bugs = {};
+
+    Bugs.init = function () {
+        handleSubmitBug();
+    };
+
+    console.log('hello');
```
Comment (vanessalin0226) [inline public/src/client/bugs.js]: Not sure if you want to keep it in here, is this just for debugging purposes?

File: src/bugs/create.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import plugins from '../plugins';
 
 type Bug = {
     bid: number,
-    title: string,
+    name: string,
     description: string,
     timestamp: string,
```
Comment (vanessalin0226) [inline src/bugs/create.ts]: Do we still need this timestamp? I think this was removed from the BugObject.yaml.

File: src/bugs/create.ts [hunk 3]
```diff
@@ -23,15 +23,17 @@ export = function (Bugs: BugsInterface) {
     Bugs.create = async function (data : Bug): Promise<number> {
         const timestamp = data.timestamp || Date.now();
```
Comment (vanessalin0226) [inline src/bugs/create.ts]: Same question about this timestamp too.

File: src/install.js [hunk 4]
```diff
@@ -492,10 +492,8 @@ async function createInitialBug() {
     console.log('Creating initial bug!');
 
     await Bugs.post({
-        bid: 1,
```
Comment (vanessalin0226) [inline src/install.js:495]: Why is this bid removed...?

**Review by vanessalin0226 — CHANGES_REQUESTED** [review]
Left some comments on some of the files, some are just questions that I'm not sure if should be deleted/ kept.

File: public/src/client/bugs.js (hunk 1 above)
Comment (Sebastian0011) [reply public/src/client/bugs.js]: Will remove

File: src/bugs/create.ts (hunk 2 above)
Comment (Sebastian0011) [reply src/bugs/create.ts]: Removed

File: src/bugs/create.ts (hunk 3 above)
Comment (Sebastian0011) [reply src/bugs/create.ts]: I still technically need `Date.now()` for adding to the sorted set if we were ever to fetch bugs from the db. It is not an actual field anymore though, you are rightt

File: src/install.js (hunk 4 above)
Comment (Sebastian0011) [reply src/install.js:495]: The db keeps track of the `nextBid` to be created, so there's no need to hardcode it here

**Review by vanessalin0226 — APPROVED** [review]

**Pull request merged**